### PR TITLE
pbs: handle poll if PBS client cannot connect

### PR DESCRIPTION
**Pull request by matthewrmshin**
If PBS `qstat` cannot connect to its server, assume that jobs managed by it are still OK.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Yes, thinking about how to write an effective test.

**Comment by matthewrmshin** [comment]
Test added.

File: lib/cylc/batch_sys_handlers/pbs.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ class PBSHandler(object):
     # N.B. The "qstat JOB_ID" command returns 1 if JOB_ID is no longer in the
     # system, so there is no need to filter its output.
     POLL_CMD = "qstat"
+    POLL_CANT_CONNECT_ERR = "Connection refused"
```
Comment (oliver-sanders) [inline lib/cylc/batch_sys_handlers/pbs.py]: How stable is this message across different `pbs` versions?

Comment (matthewrmshin) [reply lib/cylc/batch_sys_handlers/pbs.py]: We only have our own to test on our site, so no idea. Our admin team is confident that it is a good message to use though. @hjoliver may have a better insight with his connection with the people who develop PBS.

Comment (hjoliver) [reply lib/cylc/batch_sys_handlers/pbs.py]: I'll ping my PBS connections...

Comment (hjoliver) [reply lib/cylc/batch_sys_handlers/pbs.py]: (however, we can always adapt to other PBS versions in future releases if need be)

Comment (hjoliver) [reply lib/cylc/batch_sys_handlers/pbs.py]: (... pinged)

**Comment by hjoliver** [comment]
From a contact at Altair:

> " is this response standard across PBS versions?" 
That is generally correct. I've experienced this across a few versions. Also re-tested on Azure, and message is:

```
------------
Connection refused
qstat: cannot connect to server xxxxxx (errno=111)
------------
```
 
> Also correct to assume, if you get the message above, it is no reflection on jobs itself. Simply means the PBS daemons can't be contacted. As usual, PBS jobs which are already running, will keep running. Not so for array sub-jobs.
>
> But wait.... for PBS v18.x and beyond.
> - Even subjobs of array jobs, when they are running, will keep running if PBS daemons crashes or whatever.
> - Qstat will have JSON format output as an option, and customizable delimiters
> - Job's stdout/stderr can be retrieved while job is still running.
>
> .... just in case you want to plan for Cylc 8, 9, or beyond.

**Comment by matthewrmshin** [comment]
(The PBS v18.x Qstat new features are pretty exciting, but I don't think they would affect the purpose of this PR.)

**Pull request merged**